File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/playback/ItemStub.java
/**
 *
 * @file
 *
 * @brief Stub singleton implementation of Item
 *
 * @author vitamin.caig@gmail.com
 *
 */

package app.zxtune.playback;

import android.net.Uri;

import app.zxtune.Identifier;
import app.zxtune.TimeStamp;

public class ItemStub implements Item {

  private static final String EMPTY_STRING = "";
  
  @Override
  public Uri getId() {
    return Uri.EMPTY;
  }

  @Override
  public Identifier getDataId() {
    return Identifier.EMPTY;
  }

  @Override
  public String getTitle() {
    return EMPTY_STRING;
  }

  @Override
  public String getAuthor() {
    return EMPTY_STRING;
  }
  
  @Override
  public String getProgram() {
    return EMPTY_STRING;
  }
  
  @Override
  public String getComment() {
    return EMPTY_STRING;
  }

  @Override
  public String getStrings() {
    return EMPTY_STRING;
  }
  
  @Override
  public TimeStamp getDuration() {
    return TimeStamp.EMPTY;
  }

  public static Item instance() {
    return Holder.INSTANCE;
  }

  //onDemand holder idiom
  private static class Holder {
    public static final Item INSTANCE = new ItemStub();
  }  
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/playback/stubs/ItemStub.kt
/**
 *
 * @file
 *
 * @brief Stub singleton implementation of Item
 *
 * @author vitamin.caig@gmail.com
 */
package app.zxtune.playback.stubs

import android.net.Uri
import app.zxtune.TimeStamp
import app.zxtune.core.Identifier
import app.zxtune.playback.Item

open class ItemStub : Item {
    override val id: Uri
        get() = Uri.EMPTY

    override val dataId: Identifier
        get() = Identifier.EMPTY

    override val title: String
        get() = ""

    override val author: String
        get() = ""

    override val program: String
        get() = ""

    override val comment: String
        get() = ""

    override val strings: String
        get() = ""

    override val duration: TimeStamp
        get() = TimeStamp.EMPTY

    override val size: Long
        get() = 0
}
